components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+use icu_uniset::UnicodeSet;
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+
+//
+// data key structs - the structs used directly by users of data provider
+//
+
+macro_rules! data_key {
```

Comment:
I think I made this public on master, so you don't need to redefine it.

Reply:
Done.